docs/documentation/comparison_of_click_and_rich_click.md

```diff
@@ -0,0 +1,109 @@
+# Comparison of Click and rich-click
+
+**rich-click** is a [shim](https://en.wikipedia.org/wiki/Shim_(computing)) around Click,
+meaning its API is designed to mirror the Click API and intercept some of the calls to slightly different functions.
+
+Everything available via `import click` is also available via `import rich_click as click`.
+
+**rich-click** is designed to keep the additional API surface introduced on top of Click as lightweight as possible.
+
+## Click features that rich-click overrides
+
+The only things that **rich-click** explicitly overrides in the high-level API are the decorators:
+
+- `click.command()`
+- `click.group()`
+
+The only change to these decorators is that by default, their `cls=` parameters point to the **rich-click** implementations of `Command` (i.e. `RichCommand`) and `Group` (i.e. `RichGroup`).
+
+!!! info
+    There is also a thin wrapper around `pass_context()` to cast the `click.Context` type in the function signature to `click.RichContext` to assist with static type-checking with MyPy. Aside from different typing, there are no substantive changes to the `pass_context()` decorator.
+
+## Click features that rich-click does _not_ override
+
+### Base Click command classes
+
+You can still access the base Click classes by their original names:
+
+- `from rich_click import Command`
+- `from rich_click import Group`
+- `from rich_click import Context`
+
+The above are the same as importing from `click`.
+
+**rich-click**'s subclasses all have the word Rich in front of them!
+
+- `from rich_click import RichCommand`
+- `from rich_click import RichGroup`
+- `from rich_click import RichContext`
+
+### Echo and interactive elements
+
+**rich-click** deliberately does _not_ enrich certain Click features:
+
+- `click.echo()`
+- `click.echo_via_pager()`
+- `click.confirm()`
+- `click.prompt()`
+
+You are free to use these functions and they are available via `import rich_click as click`,
+but Rich's markup will not work with these functions because these functions are just the base Click implementations,
+without any changes.
+
+This is a deliberate decision that we are unlikely to change in the future.
+We do not want to maintain a more spread-out API surface, and we encourage users to become comfortable using Rich directly; it's a great library and it's worth learning a little bit about it!
+If you'd like Rich markup for your echos and interactive elements, then you can:
+
+- Replace `click.echo()` with `rich.print()`
+- Replace `click.echo_via_pager()` with `rich.Console().pager()`
+- Replace `click.confirm()` with `rich.prompt.Confirm.ask()`
+- Replace `click.prompt()` with `rich.prompt.Prompt.ask()`
+
+Below is a side-by-side comparison of Click and Rich implementations of echos and interactive elements in **rich-click**:
+
+=== "Click"
```

Comment:
Would be cool to include SVG screenshots of both under the code snippets 😉

Reply:
It's difficult to incorporate SVG screenshots because the code is interactive. It would basically just have to be faked.

Reply:
Checking it more closely I don't think it's worth it. There's not a lot to see tbh and they're visually almost identical.